modules/dxp/apps/saml/saml-opensaml-integration/src/main/java/com/liferay/saml/opensaml/integration/internal/servlet/profile/WebSsoProfileImpl.java

```diff
@@ -221,16 +220,14 @@ public void processResponse(
 					messageContext.getSubcontext(SAMLPeerEntityContext.class);
 
 				if (samlPeerEntityContext != null) {
-					String nameIdValue = Optional.ofNullable(
+					SAMLSubjectNameIdentifierContext subcontext =
```

Comment:
@jeffwu0724 Each return value of steps converted from `Optional.ofNullable/map` can be null. So we should need null checks on `subcontext` and `saml2SubjectNameID`.